### PR TITLE
Add animation tutorial

### PR DESCRIPTION
**Pull request by Esteban82**
I made this tutorial about animations. 

I will use it in the AGU24 workshop.

### CODE REVIEW

File: docs/tutorials/advance/animation.rst [hunk 1]
```diff
@@ -0,0 +1,896 @@
...
+  - **gmt begin; gmt end**: Commands to start and end a GMT script using modern syntax.
+  - **@earth_relief_06m**: A remote grid of Earth's relief with a 6-minute resolution.
+  - **-I**: Apply illumination to the grid.
+  - **-JG0/0/13c**: Perspective projection with the center at longitude 0 and latitude 0, with a 13 cm map width.
+
+
+2.3. Make the Master Frame
+===========================
+
+In this second step, I recreate the previous image but with the :gmt-module:`movie` module which is use to create animations.
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:141]:
```suggestion
In this second step, I recreate the previous image but with the :gmt-module:`movie` module which is used to create animations. 
```

File: docs/tutorials/advance/animation.rst [hunk 2]
```diff
@@ -0,0 +1,896 @@
...
+
+
+**Custom format**:
+
+- If you want another dimension, you can request a custom format directly by giving width and height and dpu (*widthxheightxdpu*).
+
+
+.. Important::
+
+  - DPU: Dots-per-unit pixel density. So it is DPI for inches or DPC for cm.
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:271]:
```suggestion
  - DPU: Dots-per-unit pixel density. So, it is DPI for inches or DPC for centimeters.
```

File: docs/tutorials/advance/animation.rst [hunk 3]
```diff
@@ -0,0 +1,896 @@
...
+
+
+2.5. Make full animation
+=========================
+
+Once the draft animation is working it is possible to increment the number of frames (-T) and movie quality (-C).
+
+In the step, I increase:
+
+- the amount of frames to 360 (``-T360``) to get the whole spin.
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:461]: Does this sound nicer?

```suggestion
- the number of frames to 360 (``-T360``) to get the whole spin.
```

File: docs/tutorials/advance/animation.rst [hunk 4]
```diff
@@ -0,0 +1,896 @@
...
+.. Tip::
+
+  Be careful. This step can be quite time (and resource) consuming.
+  By default, :gmt-module:`movie` uses all the cores available to speed up the frame creation process.
+  So probably you can't do anything else while GMT is creating all the frames (maybe you can take a break, or have lunch).
+  Also you could use `-x <https://docs.generic-mapping-tools.org/dev/gmt.html#core-full>`_ to specify the number of active cores to be used.
+
+
+3. Tutorial 2. Earthquakes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:487]:
```suggestion
3. Tutorial 2. Earthquakes
~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/tutorials/advance/animation.rst [hunk 5]
```diff
@@ -0,0 +1,896 @@
...
+
+
+..  youtube:: sagKzhI88tU
+    :align: center
+    :height: 400px
+    :aspect: 1:1
+
+
+2.5. Make full animation
+=========================
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:455]:
```suggestion
2.5. Make full animation
========================
```

File: docs/tutorials/advance/animation.rst [hunk 6]
```diff
@@ -0,0 +1,896 @@
...
+3.1. Goals of the Tutorial
+==========================
+
+- What is gmt :gmt-module:`events`.
+- How to use a background script for a movie.
+- How to enhance symbols with :gmt-module:`events`.
+
+
+3.2 Make image
+===============
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:521]:
```suggestion
3.2 Make image
==============
```

File: docs/tutorials/advance/animation.rst [hunk 7]
```diff
@@ -0,0 +1,896 @@
...
+      46.4223        -38.9126     10        260           2018-01-02T02:16:18.11
+      169.3488       -18.8355   242.77      260           2018-01-02T08:10:00.06
+      ...                                                 
+     ============== ========== ======== ================ ========================
+    - Note that the input file has the columns sorted as will be required by the :gmt-module:`plot` and :gmt-module:`events` modules. It was also used for `animation 08 <https://docs.generic-mapping-tools.org/dev//animations/anim08.html>`_. 
+    Check it to see how it was downloaded and processed.
+
+
+3.3. Make master frame
+=======================
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:556]:
```suggestion
3.3. Make master frame
======================
```

File: docs/tutorials/advance/animation.rst [hunk 8]
```diff
@@ -0,0 +1,896 @@
...
+
+
+.. Important::
+
+  - :gmt-module:`events` requires a time column in the input data and will use it and the animation time to determine when symbols should be plotted.
+  - The ``-T`` is a required argument and is used to set the current plot time.
+
+
+3.3.3. Second attempt (first frame with events)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:614]:
```suggestion
3.3.3. Second attempt (first frame with events)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/tutorials/advance/animation.rst [hunk 9]
```diff
@@ -0,0 +1,896 @@
...
+        gmt end
+        EOF
+
+        gmt movie main.sh -NQuakes -Ml,png -Zs -V -C24cx12cx80 -T2018-01-01T/2018-12-31T/1d -Gblack \
+        -Lc0 --FONT_TAG=18p,Helvetica,white --FORMAT_CLOCK_MAP=-
+
+
+
+3.4. Make draft animation
+==========================
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:669]:
```suggestion
3.4. Make draft animation
=========================
```

File: docs/tutorials/advance/animation.rst [hunk 10]
```diff
@@ -0,0 +1,896 @@
...
+
+In this step, we can make a draft animation. For this example, I recommend making a low quality (with 30 DPC) video to see if the quakes appear correctly.
+
+3.4.1. First attempt
+^^^^^^^^^^^^^^^^^^^^^
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:674]:
```suggestion
3.4.1. First attempt
^^^^^^^^^^^^^^^^^^^^
```

File: docs/tutorials/advance/animation.rst [hunk 11]
```diff
@@ -0,0 +1,896 @@
...
+    :height: 400px
+    :aspect: 2:1
+
+
+.. Warning::
+  - The above script works well but it can be more efficient if a background script is used as well.
+
+
+3.4.2. The background script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:706]:
```suggestion
3.4.2. The background script
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
3.4.2. The background script
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/tutorials/advance/animation.rst [hunk 12]
```diff
@@ -0,0 +1,896 @@
...
+    :align: center
+    :height: 400px
+    :aspect: 2:1
+
+|
+
+3.6. Make full animation with enhancement
+=========================================
+
+In the previous animation, the earthquakes appear but it is hard to see when they do it.
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:802]:
```suggestion
In the previous animation, the earthquakes appear but it is hard to see when they do it.
```

File: docs/tutorials/advance/animation.rst [hunk 13]
```diff
@@ -0,0 +1,896 @@
...
+    :height: 400px
+    :aspect: 2:1
+
+|
+
+3.6. Make full animation with enhancement
+=========================================
+
+In the previous animation, the earthquakes appear but it is hard to see when they do it. 
+With :gmt-module:`events` is possible to draw attention to the arrival of a new event.
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:803]:
```suggestion
With :gmt-module:`events` it is possible to draw attention to the arrival of a new event.
```

File: docs/tutorials/advance/animation.rst [hunk 14]
```diff
@@ -0,0 +1,896 @@
...
+
+3.6.1. How to enhance symbols with events
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:806]:
```suggestion
3.6.1. How to enhance symbols with events
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/tutorials/advance/animation.rst [hunk 15]
```diff
@@ -0,0 +1,896 @@
...
+|
+
+This can be done by using `-M <https://docs.generic-mapping-tools.org/dev/events.html#m>`_ and `-E <https://docs.generic-mapping-tools.org/dev/events.html#e>`_ arguments.
+The -M arguments allows to temporarily change attributes of the symbol like:
+ 
+- -Ms: Provide a factor to modify the size.
+- -Mc: Provide a value to brighten (up to 1) or darken (up to -1) the `color intensity <https://docs.generic-mapping-tools.org/dev/reference/colorspace.html#artificial-illumination>`_.
+- -Mt: Transparency. Set a value between 100 (invisible) to 0 (opaque).
+
+The duration of the temporary changes are control via the `-E <https://docs.generic-mapping-tools.org/dev/events.html#e>`_ argument.
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:825]:
```suggestion
The duration of the temporary changes is controlled via the `-E <https://docs.generic-mapping-tools.org/dev/events.html#e>`_ argument.
```

File: docs/tutorials/advance/animation.rst [hunk 16]
```diff
@@ -0,0 +1,896 @@
...
+
+
+.. Note::
+
+   - For finite symbols there are also *normal* and *fade* phases.
+   - It is also possible to change the data value with ``-Mv``. 
+
+
+3.6.2. Make full animation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:839]:
```suggestion
3.6.2. Make full animation
^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/tutorials/advance/animation.rst [hunk 17]
```diff
@@ -0,0 +1,896 @@
...
+~~~~~~~~~~~
+
+- The paper about animations which include explanation and examples (`Wessel et al. 2024 <https://doi.org/10.1029/2024GC011545>`_).
+
+- Check the :gmt-module:`movie` and :gmt-module:`events` modules documentation for full technical information.
+
+- See the `GMT animation gallery <https://docs.generic-mapping-tools.org/6.5/animations.html>`_ for more examples.
+
+5. References
+~~~~~~~~~~~~~
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:893]:
```suggestion
- See the `GMT animation gallery <https://docs.generic-mapping-tools.org/6.5/animations.html>`_ for more examples.


5. References
~~~~~~~~~~~~~
```

File: docs/tutorials/advance/animation.rst [hunk 18]
```diff
@@ -0,0 +1,896 @@
...
+
+.. Important::
+
+  - :gmt-module:`events` requires a time column in the input data and will use it and the animation time to determine when symbols should be plotted.
+  - The ``-T`` is a required argument and is used to set the current plot time.
+
+
+3.3.3. Second attempt (first frame with events)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Now, in this attempt I use :gmt-module:`events`. I use with ``-T${MOVIE_COL0}`` to plot the quakes as dates progresses
```
Comment (yvonnefroehlich) [inline docs/tutorials/advance/animation.rst:615]: Maybe you mean this?

```suggestion
Now, in this attempt I use :gmt-module:`events` with ``-T${MOVIE_COL0}`` to plot the quakes as dates progresses
```

**Review by yvonnefroehlich — COMMENTED** [review]
So far, just a few typo fixes.

**Comment by Esteban82** [comment]
Thanks @yvonnefroehlich for all your comments. I included them all.

File: docs/tutorials/advance/animation.rst [hunk 19]
```diff
@@ -0,0 +1,897 @@
...
+
+2.3.2. First Attempt
+^^^^^^^^^^^^^^^^^^^^^
+
+In the first attempt, I create the first frame (``-M0,png``) over a black canvas (``-Gblack``) for an HD video format (``-Chd``).
+
+     .. gmtplot::
+        :height: 400 px
+
+        cat << 'EOF' > main.sh
```
[the remark this answers is not in the document]
Comment (Esteban82) [reply docs/tutorials/advance/animation.rst:176]: I add this line 

- Bash scripting environment: The examples in this tutorial are written in Bash and may not work correctly in other shell environments (e.g., zsh, fish, or Windows cmd).

**Comment by Esteban82** [comment]
Thanks @weiji14. I include your suggestions.

**Comment by Esteban82** [comment]
I will merge this. Feel free to open a new issue if you find something wrong.

**Comment by weiji14** [comment]
Thanks @Esteban82, this looks good on https://www.generic-mapping-tools.org/gmt-examples/tutorials/advance/animation.html! We might need to make some small changes to get the special video/animation blocks to render on Jupyter Book for the AGU24 workshop, but let's do that in your PR at https://github.com/GenericMappingTools/agu24workshop/pull/16

**Pull request merged**